src/MSBuild/XMake.cs

```diff
@@ -1725,6 +1725,12 @@ internal static void GatherCommandLineSwitches(ArrayList commandLineArgs, Comman
                                 String.Equals(switchName, "maxcpucount", StringComparison.OrdinalIgnoreCase))
                             {
                                 int numberOfCpus = Environment.ProcessorCount;
+                                if (numberOfCpus == 32 && NativeMethodsShared.IsWindows) // 32-bit process, 32-bit Windows had a 32-core limit
```

Comment:
I think it would be better if either:
- This code ran always, i.e. remove the `numberOfCpus == 32` condition. It would help test the new code on more machines. Unless calling the Win32 API has a measurable perf impact.
- Or this code ran only if the process is actually 32-bit. Users may choose to run 64-bit msbuild.exe and my understanding is that this fix is not needed there.